services/manifest-repo-export-service/pkg/repository/transformer_test.go

```diff
@@ -3021,3 +3021,278 @@ func TestUndeployLogic(t *testing.T) {
+						err2 = dbHandler.DBInsertOrUpdateApplication(ctx, transaction, concreteTransformer.Application, db.AppStateChangeCreate, db.DBAppMetaData{Team: concreteTransformer.Team})
+						if err2 != nil {
+							t.Fatal(err2)
+						}
+						err2 = dbHandler.DBUpdateOrCreateRelease(ctx, transaction, db.DBReleaseWithMetaData{
+							ReleaseNumber: concreteTransformer.Version,
+							App:           concreteTransformer.Application,
+							Manifests: db.DBReleaseManifests{
+								Manifests: concreteTransformer.Manifests,
+							},
+						})
+						if err2 != nil {
+							t.Fatal(err2)
+						}
+					}
+
+					if tr.GetDBEventType() == db.EvtCreateEnvironmentLock {
+						concreteTransformer := tr.(*CreateEnvironmentLock)
+						err2 = dbHandler.DBWriteEnvironmentLock(ctx, transaction, concreteTransformer.LockId, concreteTransformer.Environment, db.LockMetadata{
+							CreatedByName:  concreteTransformer.AuthorName,
+							CreatedByEmail: concreteTransformer.AuthorEmail,
+							Message:        concreteTransformer.Message,
+							CiLink:         "", //not transported to repo
+						})
+						if err2 != nil {
+							t.Fatal(err2)
+						}
+						err2 = dbHandler.DBWriteAllEnvironmentLocks(ctx, transaction, 0, concreteTransformer.Environment, []string{concreteTransformer.LockId})
+						if err2 != nil {
+							t.Fatal(err2)
+						}
+					}
+					if tr.GetDBEventType() == db.EvtCreateUndeployApplicationVersion {
+						concreteTransformer := tr.(*CreateUndeployApplicationVersion)
+						err2 = dbHandler.DBUpdateOrCreateRelease(ctx, transaction, db.DBReleaseWithMetaData{
+							ReleaseNumber: 2,
+							App:           concreteTransformer.Application,
+							Manifests: db.DBReleaseManifests{
+								Manifests: map[string]string{ //empty manifest
+									"": "",
+								},
+							},
+							Metadata: db.DBReleaseMetaData{
+								SourceAuthor:    "",
+								SourceCommitId:  "",
+								SourceMessage:   "",
+								DisplayVersion:  "",
+								UndeployVersion: true,
+							},
+							Created: time.Now(),
+						})
+						if err2 != nil {
+							t.Fatal(err2)
+						}
+					}
+				}
+				var commitMsg []string
+				for _, t := range tc.Transformers {
+					err := repo.Apply(ctx, transaction, t)
+					if err != nil {
+						return err
+					}
+					// just for testing, we push each transformer change separately.
+					// if you need to debug this test, you can git clone the repo
+					// and we will only see anything if we push.
+					err = repo.PushRepo(ctx)
+					if err != nil {
+						return err
+					}
+				}
+				actualMsg := ""
+
+				actualMsg = repo.State().Commit.Message()
+
+				// note that we only check the LAST error here:
+				if len(commitMsg) > 0 {
+					actualMsg = commitMsg[len(commitMsg)-1]
+				}
+				if diff := cmp.Diff(tc.expectedMessage, actualMsg); diff != "" {
+					t.Errorf("commit message mismatch (-want, +got):\n%s", diff)
+				}
+
+				return nil
+			})
+
+			if err != nil {
```

Comment:
we need to also do the cmp.Diff call if err is nil, to make sure that this is exactly what we expect.